src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -540,7 +541,7 @@ public function mirror($originDir, $targetDir, \Traversable $iterator = null, $o
 
         // Iterate in destination folder to remove obsolete entries
         if ($this->exists($targetDir) && isset($options['delete']) && $options['delete']) {
-            $deleteIterator = $iterator;
+            $deleteIterator = $targetIterator;
```

Comment:
Im sorry for asking. But cloud we just do something simliar to this instead for introducing `$targetIterator`
```suggestion
            $deleteIterator = null === $iterator ? null : clone $iterator;
```

Reply:
I'm not sure I understand. It looks like the only change here is to clone `$iterator` instead of assigning it, if it's set. Does that change anything functionally?

I thought iterators all included absolute paths, which means you can't reuse an iterator across multiple parent directories. This is well beyond my comfort zone though, I could be wrong.